File: java/spring-boot-example/src/main/java/com/creolecentric/example/model/TTSJob.java
package com.creolecentric.example.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class TTSJob {
    private String id;
    private String status;
    private String text;

    @JsonProperty("voice_id")
    private String voiceId;

    @JsonProperty("model_id")
    private String modelId;

    @JsonProperty("audio_url")
    private String audioUrl;

    @JsonProperty("duration_seconds")
    private Double durationSeconds;

    @JsonProperty("credits_used")
    private Integer creditsUsed;

    @JsonProperty("created_at")
    private String createdAt;

    @JsonProperty("completed_at")
    private String completedAt;

    @JsonProperty("error_message")
    private String errorMessage;
}

File: swift/swiftui-example/Models/TTSJob.swift
import Foundation

struct TTSJob: Codable, Identifiable {
    let id: String
    let status: String
    let text: String
    let voiceId: String
    let modelId: String
    let audioUrl: String?
    let durationSeconds: Double?
    let creditsUsed: Int?
    let createdAt: String
    let completedAt: String?
    let errorMessage: String?

    enum CodingKeys: String, CodingKey {
        case id
        case status
        case text
        case voiceId = "voice_id"
        case modelId = "model_id"
        case audioUrl = "audio_url"
        case durationSeconds = "duration_seconds"
        case creditsUsed = "credits_used"
        case createdAt = "created_at"
        case completedAt = "completed_at"
        case errorMessage = "error_message"
    }
}
